src/Common/src/System/Diagnostics/Debug.cs

```diff
@@ -215,6 +268,38 @@ public static void WriteLineIf(bool condition, string value, string category)
             }
         }
 
+        private static bool NeedIndent { get; set; }
+
+        private static void WriteIndent()
+        {
+            NeedIndent = false;
+            int indentLevel = IndentLevel;
+            int indentSize = IndentSize;
+            for (int i = 0; i < indentLevel; i++)
+            {
+                switch (indentSize)
+                {
+                    case 2:
+                        WriteCore("  ");
+                        break;
+                    case 4:
+                        WriteCore("    ");
+                        break;
+                    default:
+                        for (int j = 0; j < indentSize; j++)
+                        {
+                            WriteCore(" ");
+                        }
+                        break;
+                }
```

Comment:
This doesn't look right to me.  I believe WriteCore considers each written value to be its own thing, e.g. writing it as its own entry to syslog on Unix.  Writing out each space in the indent as its own entry won't play nicely with that.